### PR TITLE
yutto: 2.0.0-beta.40 -> 2.0.0-beta.43

### PR DESCRIPTION
**Pull request by linsui**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/biliass/default.nix [hunk 1]
```diff
@@ -3,28 +3,30 @@
...
   pytestCheckHook,
 }:
 
 buildPythonPackage rec {
   pname = "biliass";
-  version = "1.3.11";
+  version = "1.3.13";
   format = "pyproject";
 
   disabled = pythonOlder "3.7";
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/biliass/default.nix:17]:
```suggestion
```
I recommend you to drop this as it is not that useful and requires to be kept up to date.

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 341333`

---
### `x86_64-linux`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>biliass (python312Packages.biliass)</li>
    <li>biliass.dist (python312Packages.biliass.dist)</li>
    <li>python311Packages.biliass</li>
    <li>python311Packages.biliass.dist</li>
    <li>yutto</li>
    <li>yutto.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>biliass (python312Packages.biliass)</li>
    <li>biliass.dist (python312Packages.biliass.dist)</li>
    <li>python311Packages.biliass</li>
    <li>python311Packages.biliass.dist</li>
    <li>yutto</li>
    <li>yutto.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>biliass (python312Packages.biliass)</li>
    <li>biliass.dist (python312Packages.biliass.dist)</li>
    <li>python311Packages.biliass</li>
    <li>python311Packages.biliass.dist</li>
    <li>yutto</li>
    <li>yutto.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>biliass (python312Packages.biliass)</li>
    <li>biliass.dist (python312Packages.biliass.dist)</li>
    <li>python311Packages.biliass</li>
    <li>python311Packages.biliass.dist</li>
    <li>yutto</li>
    <li>yutto.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Hashes are all wrong...

**Comment by linsui** [comment]
Sorry, I forgot to update the hash when I update it from .41 to .42... This depends on https://github.com/NixOS/nixpkgs/pull/344424.

File: pkgs/development/python-modules/biliass/default.nix [hunk 2]
```diff
@@ -2,29 +2,28 @@
...
+  hatchling,
+  protobuf5,
   pytestCheckHook,
 }:
 
 buildPythonPackage rec {
   pname = "biliass";
-  version = "1.3.11";
+  version = "1.3.13";
   format = "pyproject";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/biliass/default.nix:13]:
```suggestion
  pyproject = true;
```

File: pkgs/by-name/yu/yutto/package.nix [hunk 3]
```diff
@@ -8,21 +8,22 @@
 
 python3Packages.buildPythonApplication rec {
   pname = "yutto";
-  version = "2.0.0-beta.40";
+  version = "2.0.0-beta.42";
   format = "pyproject";
```
Comment (SuperSandro2000) [inline pkgs/by-name/yu/yutto/package.nix:12]:
```suggestion
  pyproject = true;
```

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 341333`

---
### `x86_64-linux`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>biliass (python312Packages.biliass)</li>
    <li>biliass.dist (python312Packages.biliass.dist)</li>
    <li>python311Packages.biliass</li>
    <li>python311Packages.biliass.dist</li>
    <li>yutto</li>
    <li>yutto.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>biliass (python312Packages.biliass)</li>
    <li>biliass.dist (python312Packages.biliass.dist)</li>
    <li>python311Packages.biliass</li>
    <li>python311Packages.biliass.dist</li>
    <li>yutto</li>
    <li>yutto.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>biliass (python312Packages.biliass)</li>
    <li>biliass.dist (python312Packages.biliass.dist)</li>
    <li>python311Packages.biliass</li>
    <li>python311Packages.biliass.dist</li>
    <li>yutto</li>
    <li>yutto.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>biliass (python312Packages.biliass)</li>
    <li>biliass.dist (python312Packages.biliass.dist)</li>
    <li>python311Packages.biliass</li>
    <li>python311Packages.biliass.dist</li>
    <li>yutto</li>
    <li>yutto.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Feel free to ping me once `protobuf` 28.2 lands in `master`.

**Comment by linsui** [comment]
@GaetanLepage  protobuf is replaced with rust binding so this can be merged now. :)

File: pkgs/by-name/yu/yutto/package.nix [hunk 4]
```diff
@@ -8,21 +8,22 @@
...
+  version = "2.0.0-beta.43";
+  pyproject = true;
 
   disabled = python3Packages.pythonOlder "3.9";
+  pythonRelaxDeps = true;
 
   src = fetchFromGitHub {
     owner = "yutto-dev";
     repo = "yutto";
     rev = "v${version}";
```
Comment (GaetanLepage) [inline pkgs/by-name/yu/yutto/package.nix:20]:
```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/development/python-modules/biliass/default.nix [hunk 5]
```diff
@@ -2,29 +2,39 @@
...
+  version = "2.0.0-beta.1";
+  pyproject = true;
 
   src = fetchFromGitHub {
     owner = "yutto-dev";
-    repo = "biliass";
-    rev = "refs/tags/v${version}";
-    hash = "sha256-hBorYAqtxTZ4LElxxJOGxC2g7sBRhRKVv6HOVHZn9FA=";
+    repo = "yutto";
+    rev = "biliass@${version}";
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/biliass/default.nix:17]:
```suggestion
    rev = "refs/tags/biliass@${version}";
```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

**Review by GaetanLepage — APPROVED** [review]
LGTM

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 341333`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>biliass (python312Packages.biliass)</li>
    <li>biliass.dist (python312Packages.biliass.dist)</li>
    <li>python311Packages.biliass</li>
    <li>python311Packages.biliass.dist</li>
    <li>yutto</li>
    <li>yutto.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>biliass (python312Packages.biliass)</li>
    <li>biliass.dist (python312Packages.biliass.dist)</li>
    <li>python311Packages.biliass</li>
    <li>python311Packages.biliass.dist</li>
    <li>yutto</li>
    <li>yutto.dist</li>
  </ul>
</details>

**Pull request merged**